salt/modules/file.py

```diff
@@ -5988,13 +6011,19 @@ def manage_file(
             or source_sum.get("hsum", __opts__["hash_type"]) != name_sum
         ):
             if not sfn:
-                sfn = __salt__["cp.cache_file"](source, saltenv, verify_ssl=verify_ssl)
+                sfn = __salt__["cp.cache_file"](
+                    source, saltenv, verify_ssl=verify_ssl, use_etag=use_etag
+                )
             if not sfn:
                 return _error(ret, "Source file '{}' not found".format(source))
             # If the downloaded file came from a non salt server or local
             # source, and we are not skipping checksum verification, then
             # verify that it matches the specified checksum.
-            if not skip_verify and urllib.parse.urlparse(source).scheme != "salt":
+            if (
+                not skip_verify
+                and urllib.parse.urlparse(source).scheme != "salt"
+                and not use_etag
```

Comment:
**question** would the DeMorgan's flip read better on here? `if not (scheme == 'salt' or skip_verify or use_etag):` :thinking:

Reply:
I tried the flip and reading it broke my brain. Part of what makes it even more confusing is looking for the same condition but with different meanings due to the negative `skip_verify` parameter combined with the positive `use_etag` parameter. Also, I added a `source` check when I refactored to move the check earlier in the function. Trying to incorporate that with the flip didn't look right to me. Leaving as-is to prevent (more) brain breakage.